### PR TITLE
implement support for generating 'swap' statements in module files

### PR DESCRIPTION
**Pull request by boegel**
useful/required for https://github.com/hpcugent/easybuild-easyblocks/pull/823

### CODE REVIEW

**Comment by boegel** [comment]
@pforai, @gppezzi: please review?

**Comment by wpoely86** [comment]
lgtm

File: test/framework/module_generator.py [hunk 1]
```diff
@@ -194,13 +194,57 @@ def test_unload(self):
...
+        else:
+            expected = '\n'.join([
+                '',
+                'swap("foo", "bar")',
+                '',
+            ])
+
+        self.assertEqual(expected, self.modgen.swap_module('foo', 'bar', guarded=False))
+
+        # guarded swap (enabled by default default)
```
Comment (gppezzi) [inline test/framework/module_generator.py]: default default?

Comment (boegel) [inline test/framework/module_generator.py]: yeah, needs another default

**Comment by boegel** [comment]
Thanks for the review @wpoely86 and @gppezzi!

**Pull request merged**